lib/pipeline.ml

```diff
@@ -89,22 +110,8 @@ let docker_specs ~analysis =
             ~analysis (`Lint `Fmt)
           :: Spec.opam_monorepo builds
       | `Opam_build selections ->
-          let selections =
-            (* TODO: This filter will be removed when a new version of debian in which capnproto has this fix "https://github.com/capnproto/capnproto/pull/1830".
-               * Related to this issue "https://github.com/mirage/capnp-rpc/issues/273" *)
-            List.filter
-              (fun s ->
-                not
-                  (Variant.arch s.Selection.variant == `I386
-                  && Variant.arch s.Selection.variant == `Aarch32
-                  && List.find_opt
-                       (fun pkg ->
-                         Astring.String.cut ~sep:"." pkg
-                         |> Option.map fst
-                         |> Option.equal String.equal (Some "conf-capnproto"))
-                       s.Selection.packages
-                     |> Option.is_some))
-              selections
+          let included_selections =
+            List.filter (fun s -> not (excluded_selection s)) selections
```

Comment:
An assumption I am making, based on https://github.com/ocurrent/ocaml-ci/pull/902 being proposed and deployed, is that this filter will take account of all of a package's dependencies, and not just direct dependencies. If this assumption is wrong, then we may still hit build failures in CI due to transitive deps.

I have spent a bit of time tracing thru the code, but it hasn't been clear to me one way or another. Any input correcting my assumption if its in error would be helpful and save some time.